src/token/HolographLegacyERC721.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.13;
+
+import "../abstract/ERC721H.sol";
+
+import "../enum/TokenUriType.sol";
+
+import "../interface/HolographERC721Interface.sol";
+import "../interface/HolographInterfacesInterface.sol";
+import "../interface/HolographInterface.sol";
+import "../interface/HolographerInterface.sol";
+
+/**
+ * @title Holograph ERC-721 Collection
+ * @author Holograph Foundation
+ * @notice A smart contract for minting and managing Holograph Bridgeable ERC721 NFTs.
+ * @dev The entire logic and functionality of the smart contract is self-contained.
+ */
+contract HolographLegacyERC721 is ERC721H {
+  /**
+   * @dev Internal reference used for minting incremental token ids.
+   */
+  uint224 private _currentTokenId;
+
+  /**
+   * @dev Enum of type of token URI to use globally for the entire contract.
+   */
+  TokenUriType private _uriType;
+
+  /**
+   * @dev Enum mapping of type of token URI to use for specific tokenId.
+   */
+  mapping(uint256 => TokenUriType) private _tokenUriType;
+
+  /**
+   * @dev Mapping of IPFS URIs for tokenIds.
+   */
+  mapping(uint256 => mapping(TokenUriType => string)) private _tokenURIs;
+
+  /**
+   * @dev Constructor is left empty and init is used instead
+   */
+  constructor() {}
+
+  /**
+   * @notice Used internally to initialize the contract instead of through a constructor
+   * @dev This function is called by the deployer/factory when creating a contract
+   * @param initPayload abi encoded payload to use for contract initilaization
+   */
+  function init(bytes memory initPayload) external override returns (bytes4) {
+    // we set this as default type since that's what Mint is currently using
+    _uriType = TokenUriType.IPFS;
+    address owner = abi.decode(initPayload, (address));
+    _setOwner(owner);
+    // run underlying initializer logic
+    return _init(initPayload);
+  }
+
+  /**
+   * @notice Get's the URI of the token.
+   * @return string The URI.
+   */
+  function tokenURI(uint256 _tokenId) external view onlyHolographer returns (string memory) {
+    TokenUriType uriType = _tokenUriType[_tokenId];
+    if (uriType == TokenUriType.UNDEFINED) {
+      uriType = _uriType;
+    }
```

Comment:
This piece of code can be refactored because it is duplicated with the lines 120-123 and 128-131

```solidity
function _getEffectiveUriType(uint256 _tokenId) internal view returns (TokenUriType) {
    TokenUriType uriType = _tokenUriType[_tokenId];
    return uriType == TokenUriType.UNDEFINED ? _uriType : uriType;
}
```

Reply:
Nice, I like that suggestion